crates/autopilot/src/domain/eth/mod.rs

```diff
@@ -0,0 +1,79 @@
+pub use primitive_types::{H160, H256, U256};
+
+/// An address. Can be an EOA or a smart contract address.
+pub type Address = SimpleValue<H160>;
+
+/// Block number.
+pub type BlockNo = SimpleValue<u64>;
+
+/// A transaction ID, AKA transaction hash.
+pub type TxId = SimpleValue<H256>;
+
+pub type TokenAddress = SimpleValue<H160>;
+pub type TokenAmount = SimpleValue<U256>;
+
+/// An amount denominated in the sell token for [`Kind::Sell`], or in
+/// the buy token for [`Kind::Buy`].
+pub type TargetAmount = SimpleValue<U256>;
+
+/// An asset on the Ethereum blockchain. Represents a particular amount of a
+/// particular token.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct Asset {
+    pub amount: TokenAmount,
+    pub token: TokenAddress,
+}
+
+/// Domain separator used for signing.
+///
+/// https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator
+#[derive(Debug, Clone, Copy)]
+pub struct DomainSeparator(pub [u8; 32]);
+
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct SimpleValue<T>(T);
```

Comment:
Yes the name is bad. The rationale is to avoid reimplementing a bunch of commonly used traits for H160, U256, u64 etc. See driver[ eth module ](https://github.com/cowprotocol/services/blob/main/crates/driver/src/domain/eth/mod.rs) as something I wanted to avoid.

Reply:
Reverted to basic types. Improving this can be done in a separate PR if needed (don't want to overkill this one)